app/public/consoleWindow.js

```diff
@@ -0,0 +1,92 @@
+const consoleWindowElement = document.getElementById("consoleWindow");
+
+const lineHolderElement = document.createElement("div");
+lineHolderElement.className = "lineHolder";
+consoleWindowElement.appendChild(lineHolderElement);
+
+const scrollElement = consoleWindowElement;
+
+const snapToBottomTolerance = 50; //When within this many pixels from the bottom of the console, new messages will cause the scroll to snap to the bottom
+let modes = ["public", "private", "important", "critical"];
+let seconds = 60*4; //for testing
+
+//Expects an object with 3 keys: time, message, classString
+function AddConsoleLine(lineObj){
+    let {time, message, classString} = lineObj;
+
+    let stringMessage = "";
+
+    if(time !== undefined){
+
+        let timeString = new Date(time * 1000).toISOString().substring(14, 19);
```

Comment:
to keep track of time we may not even need a date object. We could just get a time object sent from the backend with a minutes/seconds field and display that on the frontend. 
Could we reformat this to expect an object like {minutes: "1", seconds: "10"} and just display that?